draft-irtf-cfrg-opaque.md

```diff
@@ -615,44 +515,88 @@ struct {
         case registration_request: RegistrationRequest;
         case registration_response: RegistrationResponse;
         case registration_upload: RegistrationUpload;
-        case cred_request: CredentialRequest;
-        case cred_response: CredentialResponse;
+        case credential_request: CredentialRequest;
+        case credential_response: CredentialResponse;
     };
 } ProtocolMessage;
 ~~~
 
 Additionally, OPAQUE makes use of an additional structure `Credentials` to store
-user (client) credentials. It is structured as follows. Each public and
-private key value is an opaque byte string, specific to the AKE protocol in
-which OPAQUE is instantiated. For example, if used as raw public keys for
-TLS 1.3 {{?RFC8446}}, they may be RSA, DSA, or ECDSA keys as per {{?RFC7250}}.
+user (client) credentials. A `Credentials` structure consists of secret and
+cleartext `CredentialExtension` values. Each `CredentialExtension` indicates
+the type of extension and carries the raw bytes. This specification includes
+extensions essential for OPAQUE, including:
```

Comment:
I would not call all of them essential, just skU and pkS are.

Do we want to restrict to just these five or leave further extensibility in case it is needed (e.g., we were discussing the hardening parameters as a possible addition to the envelope).

Reply:
I noted that skU and pkS are mandatory. We might want to leave room for other things. The hardening parameters might be one of them, if they aren't transmitted at the application layer.

Reply:
(This list is meant to be extended, basically.)